product_docs/docs/pgd/5/routing/proxy.mdx

```diff
@@ -28,14 +28,25 @@ Endpoints given in the config file are used only at startup. After that, actual
 
-The `consensus_grace_period` proxy option can be used to configure the routing behaviour upon loss of a Raft leader. PGD Proxy will continue to route to the current write leader(if it is available) for this duration. If the new Raft leader is not elected during this period, the proxy will stop routing. If set to `0s` proxy will stop routing immediately.
-
 PGD Proxy responds to write leader change events that can be categorized into two modes of operation: *failover* and *switchover*.
 
 Automatic transfer of write leadership from the current write leader node to a new node in the event of Postgres or operating system crash is called failover. PGD elects a new write leader when the current write leader goes down or becomes unresponsive. Once the new write leader is elected by PGD, proxy closes existing client connections to the old write leader and redirects new client connections to the newly elected write leader.
 
 User-controlled, manual transfer of write leadership from the current write leader to a new target leader is called switchover. Switchover is triggered through the [PGD CLI switchover](../cli/command_ref/pgd_switchover) command. The command is submitted to PGD, which attempts to elect the given target node as the new write leader. Similar to failover, PGD Proxy closes existing client connections and redirects new client connections to the newly elected write leader. This is useful during server maintenance, for example, if the current write leader node needs to be stopped for maintenance like a server update or OS patch update.
 
+### Consensus Grace Period
+
+PGD Proxy provides the [consensus_grace_period](index.mdx) proxy option that can be used to configure the routing behaviour upon loss of a Raft leader. Proxy will continue to route to the current write leader(if it is available) for this duration. If the new Raft leader is not elected during this period, the proxy will stop routing. If set to `0s` proxy will stop routing immediately.
+
+The main purpose of this option is to allow users to configure the write behaviour when Raft leader is lost. When Raft leader is not present in the cluster it is not always guaranteed that the current write leader seen by the proxy is the correct one. In some cases, like network partition, example given below, it is possible that two write leaders may be seen by two different proxies attached to the same group increasing the chances of write conflicts. If this is not the expected behavior then as mentioned above grace period can be set to 0s which configures the proxy to stop routing and close existing open connections immediately when it detects Raft leader is lost.
```

Comment:
when the Raft leader is lost. When the Raft leader is not present...

If this is not the desired behavior